09-intermediate-rails/intro-to-oauth.md

```diff
@@ -1,35 +1,48 @@
+# Introduction to Authentication and OAuth
 
-Today we will introduce a technology called *OAuth*, which we will use to allow users to log into our Rails app. OAuth is short for **Open Authentication**, and it is an open standard that describes how users can use one service to log into another. If you've ever logged into a website using Facebook, you've used OAuth.
+## Learning Goals
 
+By the end of this lesson, students should be able to answer the following questions or do the following things:
 
-## Learning Goal
-By the end of this lesson, students should be able to...
+- What is _authorization_?
+- Why is authorization an interesting problem?
+- What is _OAuth_, and how do we recognize that we are using an OAuth system as a user?
+- Have exposure to OAuth's process and the following vocab:
+  - OAuth Provider
+  - User
+  - Client (as our Rails application)
+- What are trade-offs between OAuth and building our own system?
 
-- Differentiate between Authentication and Authorization
-- Describe how the OAuth protocol works at a high level
-- Identify the different roles in the OAuth workflow
-- Explain the advantages of using OAuth instead of building a custom login system
+## What is authorization?
 
-## Introduction
+We've learned about creating user login functionality in Rails using Rails' `session` hash.
+
+### Authentication != Authorization
 
-Security is a deep and nuanced field, but for our purposes there are two important pieces to get right:
 - **Authentication:** Who are you, and how can I trust you?
 - **Authorization:** Are you allowed to do that action?
 
-OAuth is an **authentication** framework.  Using OAuth a user can securely **authenticate** (log in) to a service like Github, Google or Facebook, and allow another service, like our Rails App, to access the user's profile. So users log into one service, and use that account to log into other services.
+## What is OAuth?
+
+OAuth is an **authentication** framework. OAuth is short for **Open Authentication**, and it is an open standard that describes how users can use one service to log into another.
+
+If you've ever logged into a website using Facebook, you've used OAuth. Using OAuth, a user can securely **authenticate** (log in) to one service like Github, Google or Facebook, and allow a second service, like our Rails App, to access that user's data. 
+
+So users log into one service, and use that account to log into the second service.
+
+### What is OAuth to an end user?
+
+"In product development, an _end user_ is a person who ultimately uses or is intended to ultimately use a product."
 
 If you've ever followed a link to "log in with Facebook" (or Google, Twitter, GitHub, etc) then you've used OAuth.
 
 ![Log in with Facebook button](images/oauth_in_the_wild.png)
 
-Once our Rails app has authenticated a user via OAuth, it can make decisions about whether that user is **authorized** to take various actions on the site.
+### OAuth at Introductory Detail
```

Comment:
This section no longer has any content.

Reply:
Yes! This is incomplete and not ready for review yet :) Sorry

Reply:
Hm, looking at it, I probably just mistakenly added this header anyway, so maybe @dHelmgren should delete it. I'll let her decide! Who knows what I was thinking 2 hours ago! OOP good catch tho